### ISSUE TITLE
LibreNMS with php8.1 crash at Weathermap installation

### ISSUE THREAD

**Issue by fbouynot** [librenms-plugins/Weathermap#78]
Hello,

It is the same issue than #76 but it was closed without solution given by the user.
I'm using CentOS Stream 8 with php8.1.5 from remi repo, nginx 1.14.1, MariaDB 10.3.28, Laravel 8.80.0, RRDTool 1.7.0, and LibreNMS 22.4.0-18-g70bae6605 with 2022_04_08_085504_isis_adjacencies_table_add_index (236) database schema.
If I install weathermap with LibreNMS documentation, LibreNMS fall in error state with error 500 and `ksort(): Argument #1 ($array) must be of type array, null given` log.
At the moment I delete Weathermap folder, LibreNMS is working again.
@cabello123 says to solved the issue by "adding missing arrays to the function".

Does someone know where does it come from?

**Comment by dgeo** [comment]
same here with php 8.0

**Comment by kaude101** [comment]
i have the same issue can anybody help me below the output from the cli:

`librenms@nmserver:~$ ./validate.php
====================================
Component | Version
--------- | -------
LibreNMS  | 22.7.0-9-g799c03df2
DB Schema | 2022_07_19_081224_plugins_unique_index (244)
PHP       | 8.1.2
Python    | 3.10.4
MySQL     | 10.6.7-MariaDB-2ubuntu1.1
RRDTool   | 1.7.2
SNMP      | 5.9.1
====================================

[OK]    Composer Version: 2.3.10
[OK]    Dependencies up-to-date.
[OK]    Database connection successful
[OK]    Database Schema is current
[OK]    SQL Server meets minimum requirements
[OK]    lower_case_table_names is enabled
[OK]    MySQL engine is optimal
[OK]
[OK]    Database schema correct
[OK]    MySQl and PHP time match
[OK]    rrd_dir is writable
[OK]    rrdtool version ok
[WARN]  Your install is over 24 hours out of date, last update: Fri, 22 Jul 2022 21:31:27 +0000
        [FIX]:
        Make sure your daily.sh cron is running and run ./daily.sh by hand to see if there are any errors.`

`
librenms@nmserver:~$ ./daily.sh
Updating to latest codebase                        OK
Updating Composer packages                         OK
Updated from 799c03df2 to 7071db98a                OK
Updating SQL-Schema                                OK
Updating submodules                                OK
Cleaning up DB                                     OK
Fetching notifications                             OK
Caching PeeringDB data                             OK
Caching Mac OUI data                               OK
`

i flow steps on https://docs.librenms.org/Extensions/Weathermap/
when i try to enable plugin frow webGUI i have this mesage


Whoops, looks like something went wrong. Check your librenms.log.

**Comment by LoveSkylark** [comment]
in WeatherMap.functions.php
I had to change line 1494 from:
if ($number < 0)
to:
if(!is_int($number))

or the  everything breaks when you switch form: "map style>link lablel>percentage" to "map style>link lablel>bits/sec"

**Comment by LoveSkylark** [comment]
I have working Weathermap at [LoveSkylark](https://github.com/LoveSkylark)/
[LibreNMS-Weathermap](https://github.com/LoveSkylark/LibreNMS-Weathermap)
[LibreNMS-Weathermap](https://github.com/LoveSkylark/LibreNMS-Weathermap)

**Comment by omorfin** [comment]
I found what is causing the problem in my case... 
it is when i set "Link Labels" value to "Bit/s". With "percent" or "none", it works.

**Comment by Pluppo** [comment]
> Hello, I just tested and it still doesn't work for me. I have this error with active debug :
> 
> { "message": "Unsupported operand types: string / int", "exception": "TypeError", "file": "/opt/librenms/html/plugins/Weathermap/lib/WeatherMap.functions.php", "line": 1593, "trace": [ { "file": "/opt/librenms/html/plugins/Weathermap/lib/WeatherMap.functions.php", "line": 149, "function": "nice_scalar" }, { "file": "/opt/librenms/html/plugins/Weathermap/lib/Weathermap.class.php", "line": 698, "function": "mysprintf" }, { "file": "/opt/librenms/html/plugins/Weathermap/lib/WeatherMapLink.class.php", "line": 485, "function": "ProcessString", "class": "WeatherMap", "type": "->" }, { "file": "/opt/librenms/html/plugins/Weathermap/lib/Weathermap.class.php", "line": 3403, "function": "Draw", "class": "WeatherMapLink", "type": "->" }, { "file": "/opt/librenms/html/plugins/Weathermap/editor.php", "line": 1065, "function": "DrawMap", "class": "WeatherMap", "type": "->" } ] }

I have the exact same error message when trying to edit a map using the editor. I tried changing BWLABEL in the .conf file from bits to percent, and that did let me see the map. But after that I'm unable to click on anything, none of the menu items, nodes or links can be clicked.

**Comment by LoveSkylark** [comment]
> I have the exact same error message when trying to edit a map using the editor. I tried changing BWLABEL in the .conf file from bits to percent, and that did let me see the map. But after that I'm unable to click on anything, none of the menu items, nodes or links can be clicked.

I think I got this solved try it again

**Comment by rcmcronny** [comment]
For me, this seems to work fine, map is visible and i can edit elements. So good catch.

**Comment by LoveSkylark** [comment]
> For me, this seems to work fine, map is visible and i can edit elements. So good catch.

No it broke the map, all the number are now 0 for me,  but I think the the last change fixed that.

**Comment by rcmcronny** [comment]
the commit, which makes the numbers to 0 was AFTER my comment, the change  25 min ago restores this, so i have alread the now right version :) So my comment stay valid. Thanks for your work.

**Comment by Pluppo** [comment]
> > I have the exact same error message when trying to edit a map using the editor. I tried changing BWLABEL in the .conf file from bits to percent, and that did let me see the map. But after that I'm unable to click on anything, none of the menu items, nodes or links can be clicked.
> 
> I think I got this solved try it again

It turns out the issue was related to an extension in my Chrome browser (Snow Web Application Metering), works fine without it. This extension was automatically installed by company policy without my knowledge, so it took me a bit of time before I understood why it was not working.

**Comment by fbouynot** [comment]
It works fine with https://github.com/librenms-plugins/Weathermap/pull/82

**Comment by jdsemma** [comment]
I've done a git pull so have up to date Weathermap, and after updating to php8.1 my librenms validate.php is clean. I had to edit BWLABEL to change it to "percent" before I could edit maps again. However, weathermap does not seem to be running to create images. I checked cron, and if I run map-poller.php by hand it errors:

./map-poller.php: line 1: ?php: No such file or directory
./map-poller.php: line 3: syntax error near unexpected token `('
./map-poller.php: line 3: `// Copyright (C) 2013 Neil Lathwood neil@lathwood.co.uk'
